packages/logging/src/entry.js

```diff
@@ -87,6 +114,29 @@ function Entry(metadata, data) {
     timestamp: new Date()
   }, metadata);
 
+  if (this.metadata.insertId === undefined) {
+    // JavaScript date has a very coarse granularity (millisecond), which makes
+    // it quite likely that multiple log entries would have the same timestamp.
+    // The Logging API doesn't guarantee to preserve insertion order for entries
+    // with the same timestamp. The service does use `insertId` as a secondary
+    // ordering for entries with the same timestamp. `insertId` needs to be
+    // globally unique (within the project) however.
+    //
+    // We generate an `insertId` that is going to uniquely identify the log
+    // entry across the network of services. It is composed of a ServiceId (a
+    // UUID in our case) along with an incrementing 'event counter'. The other
+    // alternative would be to use `process.hrtime()` as the event id/timestamp,
+    // but that would be more overhead.
+    insertCounter++;
+
+    // We encode in base36 to make the string compact. We left pad with zeros to
+    // make it sortable. We left pad upto a width of 13 as a max 64-bit integer
+    // would fit in 13 digits in base36.
+    var counter = leftPad(insertCounter.toString(36), 13, '0');
+    var insertId = SERVICE_ID + '-' + counter;
```

Comment:
+1. The one issue is that base64 doesn't preserve lexicographic order which we need here. Also, I think doing additions on `Uint32` would make the addition faster, and get rid of the loop.

I am starting to think it might be a good idea to split it out into a standalone module for the `EventId` concept.